synapse/api/auth.py

```diff
@@ -629,7 +629,13 @@ def get_user_by_access_token(self, token, rights="access"):
         except AuthError:
             # TODO(daniel): Remove this fallback when all existing access tokens
             # have been re-issued as macaroons.
+            if self.hs.config.expire_access_token:
+                raise AuthError(
+                    401,
+                    "access token has expired"
+                )
```

Comment:
I agree that the message in Autherror should be changed. But the problem with this function is it never raises if the access token is expired and not valid anymore, it just looks in database and returns the user anyway with the fallback. That is why I would like to raise if the access token is expired